h2o-core/src/main/java/water/DKVManager.java

```diff
@@ -122,4 +122,27 @@ protected void setupLocal() {
 
 
   }
+
+  /**
+   * Retains given keys locally, not cluster-wide.
+   *
+   * @param retainedKeys Keys to retain
+   */
+  public static void retainLocal(final Set<Key> retainedKeys) {
```

Comment:
Is this safe to be included in core? I am the current usage is harmless however, there might be hidden issues:

- Models can make assumption about the state of DKV - they can eg. assume that a specific is Frame is present in memory - then Model#remove would lead to NPE if the Frame was deleted. This is certainly a bad design but possible.

- this code might suffer from race conditions, eg. line 143 - if eg. a CV model is deleted as a part of another model (that is homed on another node), the model deletion can happen between lines 142-143 and an exception in the remove method.

Reply:
Ha! There might be a new way to do this. I just realized that. Next monday, I'll do some experiments. Thank you.

Reply:
Turns out we don't really need to do this step. Why ? Let's just clean everything after the whole test class is executed - I can't imagine any links among separate test classes.